modules/Emails/Email.php

```diff
@@ -4225,4 +4225,50 @@ protected function handleBodyInPlainTextFormat($mail)
         $mail->Body = $this->decodeDuringSend($mail->Body);
         $this->description = $mail->Body;
     }
+
+
+    public function getEmailAddressConfirmOptInTick($emailField)
```

Comment:
Similar to the message earlier. The validate fails due to the check for the number at the end. Lets move the duplicated part into an other method.

Reply:
If a validation fails with a valid data, then the validation is wrong or incomplete. It means we should fix the validation instead just remove.

Reply:
I have fixed it. I made one validation method for sugar email address field and other one for the email module. The fromSugarEmailAddressField method will call the validation method. fromSugarEmailAddressField is used by most of these the methods


I have also added this into the getConfirmOptInTickFromSugarEmailAddressField just in case the fromSugarEmailAddressField is not called.